### ISSUE TITLE
Different output for the same sequential models with dropout

### ISSUE THREAD

**Issue by eluzio** [keras-team/keras#10569]
I am building a sequential model

I built a first version:
```
def mlp0(input_dim, loss):
        model = Sequential()
        model.add(Dropout(0.5, input_shape=(input_dim,)))
        model.add(Dense(512, activation='sigmoid'))
        model.add(Dense(1, activation='relu'))
        model.compile(loss=loss, optimizer=Adagrad())
        return model
```

This gives me very good results in k-fold cross-validation, but when I predict on validation_set, the performance is bad.

So I tried another version.

```
def mlp1(input_dim, loss):
        inputs = keras.Input(shape=(input_dim,))
        x = keras.layers.Dropout(0.5)(inputs)
        x = keras.layers.Dense(512, activation='sigmoid')(x)
        outputs = keras.layers.Dense(1, activation='relu')(x)
        model = keras.Model(inputs, outputs)
        model.compile(loss=loss, optimizer=Adagrad())
        return model
```
the second model gives worst results on cross-validation but the results are compatible with the results on the validation set.

So I tried to modify the models in with L2 regularizations.

These models behave the same way:
```
def mlp0(input_dim, loss):
        model = Sequential()
        model.add(Dense(512, activation='sigmoid',  input_shape=(input_dim,), kernel_regularizer=regularizers.l2(0.01)))
        model.add(Dense(1, activation='relu', kernel_regularizer=regularizers.l2(0.01)))
        model.compile(loss=loss, optimizer=Adam())
        return model
```
```
import keras
from keras import regularizers

def mlp1(input_dim, loss):
    
        inputs = keras.Input(shape=(input_dim,))
        x = keras.layers.Dense(512, activation='sigmoid', kernel_regularizer=regularizers.l2(0.01))(inputs)
        outputs = keras.layers.Dense(1, activation='relu', kernel_regularizer=regularizers.l2(0.01))(x)
        model = keras.Model(inputs, outputs)
        model.compile(loss=loss, optimizer=Adam())
        return model
```

These make me think there is a catch in the prediction phase with the dropout.
To my eyes, the first two models are identical models except they are built in different ways. For some reason, they give me different answers. Can someone put me in the right direction?
